src/main/java/salesforce/ui/pages/product/NewProductPage.java

```diff
@@ -0,0 +1,112 @@
+package salesforce.ui.pages.product;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import salesforce.entities.Product;
+import salesforce.ui.pages.BasePage;
+import salesforce.utils.Translator;
+import java.util.HashMap;
+import java.util.Set;
+
+/**
+ * This class returns an instance of NewProductForm.
+ */
+public class NewProductPage extends BasePage {
+    @FindBy(css = ".forceActionButton:nth-child(3) > .label")
+    private WebElement saveButton;
+
+    private By productDescriptionTextArea = By.xpath("//label//span[text()='"
+            + Translator.translateValue("Products", "productDescription") + "']/../../textarea");
+    private By productFamilyComboBox = By.xpath("//span//span[text()='"
+            + Translator.translateValue("Products", "productFamily") + "']/../..//a");
+    private By activeCheckBox = By.xpath("//label//span[text()='"
+            + Translator.translateValue("Products", "active") + "']/../../input");
+
+    private static final String INPUT_XPATH = "//label//span[text()='%s']/../../input";
+
+    /**
+     * Waits for the page to load.
+     */
+    @Override
+    protected void waitForPageToLoad() {
+        getWait().until(ExpectedConditions.visibilityOf(saveButton));
+    }
+
+    /**
+     * Sets input fields.
+     *
+     * @param fieldName the name of the field
+     * @param fieldValue the value to set
+     * @return this page
+     */
+    public NewProductPage setInputField(final String fieldName, final String fieldValue) {
+        getWebElementAction().setTextField(getDriver()
+                .findElement(By.xpath(String.format(INPUT_XPATH, fieldName))), fieldValue);
+        return this;
+    }
+
+    /**
+     * Clicks on active CheckBox.
+     *
+     * @return this page
+     */
+    public NewProductPage clickActiveCheckBox() {
+        getWebElementAction().clickOnWebElement(getDriver().findElement(activeCheckBox));
+        return this;
+    }
+
+    /**
+     * Selects an option of the productFamily combo box.
+     *
+     * @param productFamilyOption the option to select
+     * @return this page
+     */
+    public NewProductPage selectProductFamilyOption(final String productFamilyOption) {
+        getWebElementAction().clickOnWebElement(getDriver().findElement(productFamilyComboBox));
+        getDriver().findElement(By.linkText(productFamilyOption)).click();
+        return this;
+    }
+
+    /**
+     * Sets the productDescription.
+     *
+     * @param productDescription to set
+     * @return this page
+     */
+    public NewProductPage setProductDescription(final String productDescription) {
```

Comment:
Does it require to return the same object? or could it be changed to a void method?

Reply:
Thanks. I have just changed that